ui/src/criteria/attribute.tsx

```diff
@@ -81,17 +129,115 @@ function AttributeEdit(props: AttributeEditProps) {
   const hintDisplayName = (hint: tanagra.EnumHintValue) =>
     hint.displayName || "Unknown Value";
 
-  const enumHintValues = underlay.entities
+  const attributeFilterHint = underlay.entities
     .find((g) => g.name === underlay.primaryEntity)
-    ?.attributes?.find((attribute) => attribute.name === props.data.attribute)
-    ?.attributeFilterHint?.enumHint?.enumHintValues;
+    ?.attributes?.find(
+      (attribute) => attribute.name === props.data.attribute
+    )?.attributeFilterHint;
+
+  const year = new Date().getFullYear();
+
+  if (isValid(attributeFilterHint?.integerBoundsHint)) {
+    const integerBoundsHint = attributeFilterHint?.integerBoundsHint;
+
+    if (
+      integerBoundsHint?.max !== null &&
+      integerBoundsHint?.max !== undefined &&
+      integerBoundsHint?.min !== null &&
+      integerBoundsHint?.min !== undefined
+    ) {
+      const min = year - integerBoundsHint?.max;
+      const max = year - integerBoundsHint.min;
+
+      const [minValue, setMinValue] = React.useState<number>(min);
+      const [maxValue, setMaxValue] = React.useState<number>(max);
+
+      const updateValue = (newMin: number, newMax: number) => {
+        setMinValue(newMin);
+        setMaxValue(newMax);
+        props.dispatchFn(
+          produce(props.data, (data) => {
+            data.min = newMin;
+            data.max = newMax;
+          })
+        );
+      };
+
+      React.useEffect(() => updateValue(min, max), []);
+
+      const handleChange = (event: Event, newValue: number | number[]) => {
```

Comment:
Why does `newValue` have `number | number[]` type when you immediate force it to be `number[]` on the next line?

Reply:
because the newValue type is defined by this slider package.

Reply:
Because the slider onChange props requires a function signature `function(event: Event, value: number | Array<number>, activeThumb: number) => void` according to https://mui.com/material-ui/api/slider/#props. However, I am using a range slider which always has two value, so it can be forced to be number[] (according to the source code of range slider example in https://mui.com/material-ui/react-slider/#range-slider).